### ISSUE TITLE
[GRPC] Error message lost in v1.10 

### ISSUE THREAD

**Issue by chenjpu** [dapr/dapr#6055]
## In what area(s)?

> /area runtime

## What version of Dapr?

> 1.10.2

## Expected Behavior

```
== APP == err: rpc error: code = Unimplemented desc = method echoHandler not implemented
```

## Actual Behavior

```
== APP == err: rpc error: code = Internal desc = fail to invoke, id: serving, err: rpc error: code = Unknown desc = message is nil
```

## Steps to Reproduce the Problem

server
```
package main

import (
	"context"
	"google.golang.org/grpc/codes"
	"google.golang.org/grpc/status"
	"log"

	"github.com/dapr/go-sdk/service/common"
	daprd "github.com/dapr/go-sdk/service/grpc"
)

func main() {
	// create a Dapr service server
	s, err := daprd.NewService(":50001")
	if err != nil {
		log.Fatalf("failed to start the server: %v", err)
	}
	// add a service to service invocation handler
	if err := s.AddServiceInvocationHandler("echo", echoHandler); err != nil {
		log.Fatalf("error adding invocation handler: %v", err)
	}
	// start the server
	if err := s.Start(); err != nil {
		log.Fatalf("server error: %v", err)
	}
}

func echoHandler(ctx context.Context, in *common.InvocationEvent) (out *common.Content, err error) {
	return nil, status.Errorf(codes.Unimplemented, "method echoHandler not implemented")
}

```

client

```
package main

import (
	"context"
	"fmt"
	dapr "github.com/dapr/go-sdk/client"
)

func main() {
	// just for this demo
	ctx := context.Background()
	// create the client
	client, err := dapr.NewClient()
	if err != nil {
		panic(err)
	}
	defer client.Close()
	// invoke a method called EchoMethod on another dapr enabled service
	content := &dapr.DataContent{
		ContentType: "text/plain",
		Data:        []byte("hellow"),
	}
	resp, err := client.InvokeMethodWithContent(ctx, "serving", "echo", "post", content)
	if err != nil {
		fmt.Printf("err: %s", err)
		return
	}
	fmt.Printf("service method invoked, response: %s\n", string(resp))
}
```

**Comment by chenjpu** [comment]
https://github.com/dapr/dapr/blob/d02fb79c051f8d7d45097d5eb06b3153ce4a3a24/pkg/messaging/v1/invoke_method_response.go#L161-L163

**Comment by yaron2** [comment]
Putting P0 for now. cc @ItalyPaleAle

**Comment by ryou90** [comment]
Yes, I can confirm this Behaviour since update from 1.9.X to 1.10.2.
In my case I create protobuf messages for communication. But when the messages are empty, there are lost. In 1.9.X this works without problems.

**Comment by Taction** [comment]
I found the reason, if no one is currently working on this, I'll submit a pr.

**Comment by yaron2** [comment]
> I found the reason, if no one is currently working on this, I'll submit a pr.

Thank you!

**Comment by ItalyPaleAle** [comment]
Fixed in release-1.10, will be released in Dapr 1.10.3
